src/transpilation/plugins.ts

```diff
@@ -45,6 +45,8 @@ export interface Plugin {
         emitHost: EmitHost,
         result: EmitFile[]
     ) => ts.Diagnostic[] | void;
+
+    onImportResolutionFailure?: (packageRoot : string, dependency : string) => string | undefined
```

Comment:
A module resolution plugin is something we have long been considering, and I think a better interface would be something like:

```ts
interface ModuleResolutionRequest {
    moduleIdentifier: string; // "foo.bar.baz" from require("foo.bar.baz")
    requiringFile: string; // path of the (lua) file containing the require
    // maybe some other stuff I'm forgetting now?
    // maybe a 'defaultResolve: (ModuleResolutionRequest) => string' to call the default behavior?
    // Not sure why anyone would want that
}

moduleResolution?: (moduleToBeResolved: ModuleResolutionRequest) => string | undefined
```

Then crucially this plugin would be executed _before_ our regular module resolution process, allowing users to intercept before it even starts. If it returns undefined it will go into the default behavior. I'm just wondering if we want to somehow maybe also be able to indicate that a specific import should _not_ be emitted, but maybe that's for a future version.

Reply:
I think this interface makes more sense, it gives more control to someone writing a plugin to identify the package they wish to use via the module identifier. Ill start working on it.